### PR TITLE
Restore calls to self.block_on in cosmos chain impl

### PR DESCRIPTION
**Pull request by romac**
Closes: #537

## Description

Restore calls to `block_on` on the Tokio runtime that is passed around in the relayer.

Since Tokio 1.0, the performance issues we were seeing previously seem to have vanished, likely due
to general perf improvements in Tokio but also due to the fact that one can now call `block_on` on a Tokio
runtime without having a mutable reference to it, and thus avoids having to access it through `Mutex`.

This PR also introduces a `time!` macro, which measures the time spent in the scope where the macro is called.
This is helpful to quickly figure out how much time a block of code takes to execute.
This macro only has an effect when the `profiling` feature of the `relayer-cli` is enabled.
One should thus compile/run the relayer with the `--features=profiling` flag in order to see any output.

______

For contributor use:

- [ ] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [ ] If applicable: Unit tests written, added test to CI.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Comment by romac** [comment]
Screen Shot 2021-01-19 at 18 54 59

**Review by ancazamfir — APPROVED** [review]
Looks good! Love the profiling feature. Interested to see those logs/ elapsed times with the testnets.

File: relayer/src/chain/cosmos.rs [hunk 1]
```diff
@@ -116,21 +119,29 @@ impl CosmosSDKChain {
...
+        Ok(self
+            .block_on(self.rpc_client().genesis())?
             .map_err(|e| Kind::Rpc.context(e))?
             .consensus_params)
     }
 
     /// Run a future to completion on the Tokio runtime.
     fn block_on<F: Future>(&self, f: F) -> Result<F::Output, Error> {
+        crate::time!("block_on");
+        // Ok(crate::util::block_on(f))
```
Comment (ancazamfir) [inline relayer/src/chain/cosmos.rs:133]:
```suggestion
```
remove?

**Review by adizere — COMMENTED** [review]
Can we also get a changelog update?

File: relayer-cli/relayer_operation_instructions.md [hunk 2]
```diff
@@ -1,136 +1,184 @@
...
-```
-e.g.:
-```shell script
-./dev-env my_config.toml ibc-0 ibc-1
-```
+1. Clone gaia:
+
+    ```shell script
+    git clone https://github.com/cosmos/gaia.git ~/go/src/github.com/comsos/gaia
+    ~/go/src/github.com/comsos/gaia ; git co new-main ; make install
```
Comment (ancazamfir) [inline relayer-cli/relayer_operation_instructions.md:9]:
```suggestion
    ~/go/src/github.com/comsos/gaia ; git co v3.0.0 ; make install
```

**Review by ancazamfir — APPROVED** [review]
Looks good, thanks for the cleanup!

**Pull request merged**